tsconfig.json

```diff
@@ -7,7 +7,16 @@
             "es7"
         ],
         "sourceMap": true,
-        "rootDir": "."
+        "rootDir": ".",
+        "alwaysStrict": true,
+        "noFallthroughCasesInSwitch": true,
+        "noImplicitAny": false, // TODO
+        "noImplicitReturns": false, // TODO
+        "noImplicitThis": true,
+        "noUnusedLocals": false, // TODO
+        "noUnusedParameters": false, // TODO
+        "strictNullChecks": false, // TODO
```

Comment:
It feels like there are more settings here than other repos. Are they helpful? Should we add them to other repos?